src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java

```diff
@@ -0,0 +1,282 @@
+				Long htpSampleId = externalDbEntity.getId();
+				Map<String, Object> params = new HashMap<>();
+				params.put("htpExpressionSample.id", htpSampleId);
+				SearchResponse<HTPExpressionDatasetSampleAnnotation> searchResponse = htpExpressionDatasetSampleAnnotationDAO.findByParams(params);
+				if (searchResponse == null || searchResponse.getSingleResult() == null) {
+					htpSampleAnnotation = new HTPExpressionDatasetSampleAnnotation();
+					htpSampleAnnotation.setHtpExpressionSample(externalDbEntity);
+				} else {
+					htpSampleAnnotation = searchResponse.getSingleResult();
+				}
+			} else {
+				htpSampleAnnotationResponse.addErrorMessage("SampleId", ValidationConstants.INVALID_MESSAGE + " (" + curie + ")");
+				htpSampleAnnotation = new HTPExpressionDatasetSampleAnnotation();
+			}
+		} else {
+			htpSampleAnnotationResponse.addErrorMessage("SampleId", ValidationConstants.REQUIRED_MESSAGE);
+			htpSampleAnnotation = new HTPExpressionDatasetSampleAnnotation();
+		}
+
+		if (StringUtils.isNotEmpty(dto.getSampleTitle())) {
+			htpSampleAnnotation.setHtpExpressionSampleTitle(dto.getSampleTitle());
+		}
+
+		if (StringUtils.isNotEmpty(dto.getAbundance())) {
+			htpSampleAnnotation.setAbundance(dto.getAbundance());
+		}
+		
+		if (StringUtils.isNotEmpty(dto.getSampleType())) {
+			String curie = dto.getSampleType();
+			OBITerm obiTerm = obiTermService.findByCurie(curie);
+			if (obiTerm != null) {
+				htpSampleAnnotation.setHtpExpressionSampleType(obiTerm);
+			} else {
+				htpSampleAnnotationResponse.addErrorMessage("SampleType", ValidationConstants.INVALID_MESSAGE + " (" + curie + ")");
+			}
+		} else {
+			htpSampleAnnotationResponse.addErrorMessage("SampleType", ValidationConstants.REQUIRED_MESSAGE);
+		}
+
+		if (dto.getSampleAge() != null) {
+			if (htpSampleAnnotation.getHtpExpressionSampleAge() == null) {
+				htpSampleAnnotation.setHtpExpressionSampleAge(new BioSampleAge());
+			}
+			ObjectResponse<TemporalContext> temporalContextObjectResponse = geneExpressionAnnotationFmsDTOValidator.validateTemporalContext(dto.getSampleAge().getStage());
+			if (temporalContextObjectResponse.hasErrors()) {
+				htpSampleAnnotationResponse.addErrorMessage("Sample Age - Stage", temporalContextObjectResponse.errorMessagesString());
+			} else {
+				TemporalContext temporalContext = geneExpressionAnnotationFmsDTOValidator.updateTemporalContext(temporalContextObjectResponse, htpSampleAnnotation.getHtpExpressionSampleAge().getStage());
+				htpSampleAnnotation.getHtpExpressionSampleAge().setAge(dto.getSampleAge().getAge());
+				htpSampleAnnotation.getHtpExpressionSampleAge().setWhenExpressedStageName(dto.getSampleAge().getStage().getStageName());
+				htpSampleAnnotation.getHtpExpressionSampleAge().setStage(temporalContext);
+			}
+		}
+
+		List<Long> idsToRemove = new ArrayList<>();
+		if (CollectionUtils.isNotEmpty(dto.getSampleLocations())) {
+			if(CollectionUtils.isNotEmpty(htpSampleAnnotation.getHtpExpressionSampleLocations())) {
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.
<!-- __reviewdog__:ChA0N2M0MGY4YzlkNzE0NGNkEgpjaGVja3N0eWxl -->